lib/foreman_snapshot_management/engine.rb

```diff
@@ -99,4 +99,13 @@ class Engine < ::Rails::Engine
       Foreman::Gettext::Support.add_text_domain locale_domain, locale_dir
     end
   end
+
+  def self.fog_vsphere_namespace
+    require 'fog/vsphere/version'
```

Comment:
Is it common to require this inside the function? Would it hurt in the module scope?

Reply:
I wanted to have it here so it's guarded by `if Foreman::Model::Vmware.available?`. But this whole thing is quite a hack. Ideally, we'd just support one schema of the module or stop monkey patching fog-vsphere at all.

Reply:
I do not know the implications of `require`, but can we have it inside the `||=` to not call it every time? Then i think we are good to go.

Reply:
Done. I extracted the "calculation" to a dedicated method.